pkg/boot/linux.go

```diff
@@ -29,13 +29,29 @@ type LinuxImage struct {
 	Cmdline     string
 	BootRank    int
 	LoadSyscall bool
-	DeviceTree  io.ReaderAt
 
 	KexecOpts linux.KexecOptions
 }
 
+// LoadedLinuxImage is a processed version of LinuxImage.
+//
+// Main difference being that kernel and initrd is made as
+// a read-only *os.File. There is also additional processing
+// such as DTB, if available under KexecOpts, will be appended
+// to Initrd.
+type LoadedLinuxImage struct {
+	Name        string
+	Kernel      *os.File
+	Initrd      *os.File
+	Cmdline     string
+	LoadSyscall bool
+	KexecOpts   linux.KexecOptions
+}
+
 var _ OSImage = &LinuxImage{}
 
+var errNilKernel = errors.New("linuxImage.Kernel must be non-nil")
```

Comment:
This kind of message is more about internal implementation and less about giving the poor user some information about how to correct a problem. Instead of this message, can you provide something that might tell them what to do?

Reply:
suggestion ? -- I feel this is pretty clear, perhaps the dot syntax is a bit weird ;0

Reply:
removing the doc, and use plain english instead